### DISCUSSION TITLE
What tool should we use for Authorization and roles?

### DISCUSSION THREAD

**Discussion by wassson** [caley-io/email-rails#88] [Q&A]
With rails we have several good options - CanCanCan, Pundit, and ActionPolicy. What have you used, and what is your opinion on what this project should move forward with?

**Comment by sriram2520** [comment]
I have used CanCanCan before and it has served me well in the past. But it tends to get muddy over the long run with long class for abilities.
I see more rails devs moving towards ActionPolicy and would be eager to hear thoughts from others.

**Reply by wassson** [reply]
That's exactly my thought. CanCanCan (CCC?) is awesome until your schema gets bigger (which ours is likely to grow quite a bit). Would be very interested in ActionPolicy

**Reply by metronom72** [reply]
hmm, it depends, cancan more about role based model. but pundit allow more flexible logic, plus pundit has better integration with third party gems

i tried to used action policy, but it depends on the integration that you are trying to do

**Reply by wassson** [reply]
Good feedback. It's definitely important that we spec out the roles. Realistically, we're just looking at the ability to create teams within a workspace, or add users to a workspace or a team respectively. I think Action Policy should be more than enough for now
